### DISCUSSION TITLE
Alist似乎会自动上传本地文件？

### DISCUSSION THREAD

**Discussion by Nielson-cn** [AlistGo/alist#3230] [General]
Alist版本：3.91（试了其它版本也一样）

今天偶然发现Alist在后台有流量。在Alist命令行窗口发现有人在访问我之前下载的AMD驱动安装文件和电影……

![捕获2](https://user-images.githubusercontent.com/99910815/216283825-0c663227-f8db-4c7d-8362-f47f51e60a32.PNG)

![捕获](https://user-images.githubusercontent.com/99910815/216283729-c05231aa-9233-4a14-967c-564d4fcdafa5.PNG)

![捕获3](https://user-images.githubusercontent.com/99910815/216283831-58bb95df-26e6-4aa4-a1f6-08a1752bc4a1.PNG)

还有一个已经删除的电影，也有人访问——Alist命令行窗口显示“找不到”。

不知各位有没有同样情况？

**Comment by Nielson-cn** [comment]
补充：

1、Alist只有我一个人用（私人NAS）。内置的管理员账号已改名并设置了强密码。而且更换密码后重启Alist仅几秒钟就又有人访问下载……

2、Alist的guest账号无法删除，但已经改名且设了强密码（乱密码，我都不知道是啥……），且同样是换密码后重启Alist不影响外人访问。

3、这些文件并非通过Alist的Aira2下载（我没有配置RPC），而是用其它方式下载后放在Alist可以访问的文件夹里。那部《北平无战事》甚至不在我下载时的文件夹里（迅雷下载到Downloads，后通过资源管理器移动到“剧集”）。

4、后台没有开迅雷或qBittorent之类的软件——即使开了也不会是从Alist走流量吧……

**Comment by yueboke** [comment]
正准备安装呢，如果程序有后门可不敢用

**Reply by Nielson-cn** [reply]
从作者的回复看，应该不是“后门”，而是我对“列表应用”的理解有偏差。

**Comment by xhofe** [comment]
`偷偷` 这个词挺令人心寒的，对一个完全开源的项目来说。

**Reply by Nielson-cn** [reply]
水平不够，看不懂代码…… -_-!!
但“未经允许就上传本地文件”且“没有请求允许的提示”；如果不打开命令行窗口就看不到上传——甚至早期版本（我之前用的好像是3.6，前两天才升级到3.91）在命令行窗口都看不到“有上传”而只能在log文件里看到。这说是“偷偷”应该不过分吧……

**Reply by xhofe** [reply]
- 首先，请解释一下什么叫`未经允许就上传本地文件`，上传到了哪里？
- 其次，什么叫`没有请求允许的提示`？请理解这是一个什么应用；
- 最后，在log文件中记录日志是一件很正常的事情，在标准输出中输出日志只是为了方便调试，很多应用都不会往标准输出写日志。

**Reply by Nielson-cn** [reply]
> * 首先，请解释一下什么叫`未经允许就上传本地文件`，上传到了哪里？
> * 其次，什么叫`没有请求允许的提示`？请理解这是一个什么应用；
> * 最后，在log文件中记录日志是一件很正常的事情，在标准输出中输出日志只是为了方便调试，很多应用都不会往标准输出写日志。

1、“未经允许就上传”是指：没有提示我“会把我硬盘里的文件上传给别人”。上传到哪里我也不知道——从命令行窗口看，我那个AMD显卡驱动被上传给了IP地址222.174.176.78的人（我不知道他是谁）；《北平无战事》被上传给了IP地址222.216.117.95的人（我也不知道是谁）。

2、我理解Alist是这样的应用：把我的文件（硬盘里的、网盘里的）汇总到一起，我和我许可的用户可以远程访问。但我提供的截图显示，有人未经我许可却可以获得我硬盘里的文件。而在我看来，既然Alist没有说自己是P2P软件，那如果有人没有我给的账号密码，却想通过Alist访问我硬盘里的文件，Alist应该拒绝或问我“是否允许”。比如迅雷在安装协议里会提醒我会有P2P分享，BT软件的控制界面可以选择“是否允许上传（做种）”。

3、log记录日志、标准输出里是否输出日志，这些我可以理解。我只是觉得：没告诉我“会把硬盘里的文件上传给别人”却事实上做了这件事，这种行为用“偷偷”二字应该没什么错误。

**Reply by xhofe** [reply]
- 首先， alist的定义是一个列表程序，所以它本身就是公开给任何人的，所以它默认存在一个guest账户（即未登录）如果你没有给文件夹添加任何密码那么默认所有的人都可以访问你的所有文件；
- 如果不想给所有人访问你应该为它添加一个[密码](https://alist.nn.ci/guide/advanced/meta.html#password) 而不是提示所谓的`是否允许`，你还可以将guest用户的base_path设置为空文件夹来阻止guest的访问；
- `会把硬盘里的文件上传给别人`这是一个列表应用的常识，也是最主要的功能，如果你认为这是`偷偷`，那么你可以选择不使用本软件。

**Reply by Nielson-cn** [reply]
哦……这么说来，我对“列表程序”的理解可能太外行了……未登录guest也可以访问下载我放进列表程序的所有文件，这恐怕不太符合一般用户的常识…… 毕竟“列表”最多有“展示”的意思，并不包含“分享内容”……-_-!!!

另外，我刚才在“元信息”里给root文件夹设了一下密码，结果发现登出后不能再登入了……网页一直不出现登录界面（不断转圈圈）。可能是我设置不对，我再学习一下……

**Reply by Nielson-cn** [reply]
仍然有两个疑问：

第一个：
刚才我把Alist删除重装了，可以正常登录，但没有任何“存储”（这很正常）。不过这种情况下，命令行窗口却显示又有人获取了我硬盘里的文件……该文件处于“剧集”文件夹，是我【之前】放进Alist“本机存储”的，但删除重装后还没有放进“存储”啊……为啥有人可以访问？

以下截图是我已经设置了“存储”后重新启动Alist后的情况。可以看到我放进“存储”里的并没有Storage文件夹。但外界仍然可以访问Storage里的文件。
![捕获4](https://user-images.githubusercontent.com/99910815/216366569-8587a02b-7450-485a-bf3b-25fa4eb59329.PNG)


第二个：
“列表应用”是把我放进“存储”的那些文件夹里的文件公开给别人，对吗？但我有一个之前下载的电影文件已经删除了，为什么还在命令行窗口里看到有人试图访问该文件（Alist显示“找不到”）呢？而且这同样发生在刚才——我删除Alist重装后。@_@

**Reply by xhofe** [reply]
这只是别人的访问日志记录，你无法控制别人访问什么路径。可能别人之前保存了这个地址再次打开了，对于不存在的存储它只会得到一个错误码，不会真的返回文件。

**Reply by Nielson-cn** [reply]
> 这只是别人的访问日志记录，你无法控制别人访问什么路径。可能别人之前保存了这个地址再次打开了，对于不存在的存储它只会得到一个错误码，不会真的返回文件。

谢谢说明。

那如果该存储存在，但我已经把guest指向了空文件夹，别人（没有账号密码）还能通过已保存的路径访问该存储里的文件吗？

**Reply by xhofe** [reply]
不可以，我想我已经说的很清楚，这条日志只是一个访问记录，访问已经禁用/删除/不存在的页面，只会得到一个错误。像这样：
![image](https://user-images.githubusercontent.com/36558727/216519945-f2494083-913c-452f-8eb8-409875f1ad47.png)

**Reply by Nielson-cn** [reply]
> 不可以，我想我已经说的很清楚，这条日志只是一个访问记录，访问已经禁用/删除/不存在的页面，只会得到一个错误。像这样： ![image](https://user-images.githubusercontent.com/36558727/216519945-f2494083-913c-452f-8eb8-409875f1ad47.png)

明白了。非常感谢！

之前不知道“列表应用”会允许【未登录者】获取存储里的文件，以为把Guest设个强密码就够了。现已把问题中“偷偷”的说法改了。还请原谅。

另外，看到有其他人也对“不能删除guest账户”有疑虑。所以有点好奇：为什么Alist会默认允许【没有登录】（连guest都没登录）的人访问存储？这个设计很反常识，尤其是浏览器打开网址出现的是登录页面，很难想到不登录居然也可以看到里面的内容……如果方便，不知道您能否为我科普一下这种设计的用意？谢谢！

**Reply by xhofe** [reply]
不是连guest都没登录，而是没有登录时的身份就是Guest。

如果你用过其他的列表程序或者更早期的版本，他们是没有用户这个功能的，alist只是后来在一般的列表程序上加入了【用户】的功能。

Guest作为一个特殊的用户角色，即`游客`，指的就是未登录的用户，所以Guest的密码是无效的。

**Reply by Nielson-cn** [reply]
> 不是连guest都没登录，而是没有登录时的身份就是Guest。
> 
> 如果你用过其他的列表程序或者更早期的版本，他们是没有用户这个功能的，alist只是后来在一般的列表程序上加入了【用户】的功能。
> 
> Guest作为一个特殊的用户角色，即`游客`，指的就是未登录的用户，所以Guest的密码是无效的。

“Guest的密码是无效的”——天！第一次用列表程序，这设置真是完全想不到……一直以来我玩的软件guest要么可以设置密码（如Windows），要么干脆不允许设置密码（讲明了是让未注册的人浏览的），如一些论坛程序。“可以设置密码，但其实密码无效”，这设置实在有点反常识（测试了一下，不仅改密码无效，连Guest改名也是允许但无效）…… -_-!!! 

看到有其它用户也在问“允许删除guest“的问题，建议还是把这些情况写到软件说明里比较好。希望Alist越做越好！

**Reply by H3nryM4** [reply]
> > 不是连guest都没登录，而是没有登录时的身份就是Guest。
> > 如果你用过其他的列表程序或者更早期的版本，他们是没有用户这个功能的，alist只是后来在一般的列表程序上加入了【用户】的功能。
> > Guest作为一个特殊的用户角色，即`游客`，指的就是未登录的用户，所以Guest的密码是无效的。
> 
> “Guest的密码是无效的”——天！第一次用列表程序，这设置真是完全想不到……一直以来我玩的软件guest要么可以设置密码（如Windows），要么干脆不允许设置密码（讲明了是让未注册的人浏览的），如一些论坛程序。“可以设置密码，但其实密码无效”，这设置实在有点反常识（测试了一下，不仅改密码无效，连Guest改名也是允许但无效）…… -_-!!!
> 
> 看到有其它用户也在问“允许删除guest“的问题，建议还是把这些情况写到软件说明里比较好。希望Alist越做越好！

作者的脾气太好了。直接一句 RTFM。

![屏幕截图 2023-02-05 111659](https://user-images.githubusercontent.com/53041402/216799696-2a9169b7-8edc-4e17-bfdb-99138435291b.png)

**Reply by Nielson-cn** [reply]
反常识的东西，不应以“手册里写了”来把责任推给用户。

**Reply by H3nryM4** [reply]
> 反常识的东西，不应以“手册里写了”来把责任推给用户。

开源社区的常识就是手册，而不是把脑子里想当然的东西叫做所谓的“常识”。
同时，开源社区、开源项目以及开发者，没有任何责任、义务来为你自以为的“用户”做服务。
反而，现在的情况是出了问题后把“责任”推给已经详细写明使用手册的、没有收你任何服务费用的开发者。

[《提问的智慧》](https://github.com/ryanhanwu/How-To-Ask-Questions-The-Smart-Way/blob/main/README-zh_CN.md)是你应该阅读的文章。
![image](https://user-images.githubusercontent.com/53041402/216823938-c679088c-4ffc-4c98-948a-22fdd9b3e154.png)

**Reply by Nielson-cn** [reply]
谢谢~《提问的智慧》我也经常推荐给别人。而从我另一个提问“映射网络驱动器无法写入百度网盘和天翼云”里也可以看出我并不是“提问前完全不做努力”的人。

但是，“常识”的意思本就是“不用查资料就知道的知识”。不同软件作者编写的手册并不统一，所以如果必须查阅“该软件自己的手册才能知道”，那这就没有资格被称为“常识”。而“反常识”的东西，即使写进了手册，也不成为指责使用者“没有看”“想当然”的理由。

至于“开源软件开发者没有义务为用户做服务”“出了问题把责任推给开发者”……我想你忽略了一件事：提问者【本就不曾“要求”】开发者做出回答。如果不想回答大可置之不理甚至删帖（不知道Github有没有这个功能），或者像你说的回复RTFM也不失为一种选择。但是，正如提问者如果问出愚蠢的问题会降低他在旁人眼中的评价，如果提问者的问题存在其他人的共鸣，那开发者的态度也会影响别人如何评价他（和他的软件）。

最后，也建议你再去读一遍《提问的智慧》，好好体会一下它的主旨是“用来帮助提问者更好地获得答案”还是“用来批评提问者做得还不够好”。

**Comment by edgar88** [comment]
@Xhofe 不是很明白这里说的“列表应用”会允许【未登录者】获取存储里的文件，那岂不是意味着我挂载的网盘的资料谁都可以看的到了？那这个隐私和安全问题岂不是非常大？

**Comment by xhofe** [comment]
在此线程上我不想再做过多的解释，统一回复：
- 默认公开，除非自行在元信息中添加密码
- 参考#3018 中的方法阻止Guest的访问
- 在未来的版本中会增加禁用Guest用户
